src/selectors/vaccine.js

```diff
@@ -21,3 +21,27 @@ export const selectIsSyncingTemps = ({ vaccine }) => {
   const { isSyncingTemps = false } = vaccine || {};
   return isSyncingTemps;
 };
+
+export const selectDownloadingLogsFrom = ({ vaccine }) => {
+  const { downloadingLogsFrom } = vaccine;
+  return downloadingLogsFrom;
+};
+
+export const selectLastDownloadTime = ({ vaccine }, macAddress) => {
+  const { lastDownloadTime } = vaccine;
+  return lastDownloadTime[macAddress] ? new Date(lastDownloadTime[macAddress]) : null;
+};
+
+export const selectLastDownloadFailed = ({ vaccine }, macAddress) => {
+  const { lastDownloadStatus } = vaccine;
+  const status = lastDownloadStatus[macAddress];
+  const lastDownloadFailed = status === 'ERROR';
```

Comment:
or `status !== 'OK'` to allow for other error states?

Reply:
Nice! 
1f417e7